src/scripts/generate-heading-ids.js

```diff
@@ -57,10 +59,22 @@ const addHeaderID = (line, slugger, write = false) => {
     .replace(/\{#[^}]+\}/, "")
     .trim()
   const headingLevel = line.slice(0, line.indexOf(" "))
+
+  curMaxLevel =
+    headingLevel.length > curMaxLevel ? headingLevel.length : curMaxLevel
+
+  const headingLevelArrayIndex = headingLevel.length - 1
+  if (
+    curLevel[headingLevelArrayIndex] == undefined ||
+    curLevel[headingLevelArrayIndex] == null
+  ) {
+    curLevel[headingLevelArrayIndex] = 0
+  }
   curLevel[headingLevel.length - 1]++
```

Comment:
```suggestion
  curLevel[headingLevelArrayIndex]++
```
Should we use `headingLevelArrayIndex` her now that variable exists?

Reply:
Indeed we should